src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,243 @@
+package duke;
+
+import java.time.LocalDate;
+
+import java.util.Arrays;
+import java.util.ArrayList;
+
+import java.lang.StringBuilder;
+
+/**
+ * Represents a parser that can read the input and execute the corresponding
+ * command
+ */
+
+public class Parser {
+
+    /**
+     * check if input string has less than 2 words *
+     * 
+     * @param splitInput Array of words in input string
+     * @return true if splitInput < 2 else false
+     */
+
+    private static boolean checkDescription(String[] splitInput) {
+        return splitInput.length < 2;
+    }
+
+    /**
+     * convert input string to LocalDate type
+     * 
+     * @param splitInput Array of words in input string
+     * @return LocalDate object of parsed input string else null
+     */
+
+    private static LocalDate getDate(String[] splitInput) {
+        if (splitInput.length >= 4) {
+            if (splitInput[splitInput.length - 2].equals("/by")) {
+                try {
+                    // in form yyyy-mm-dd
+                    LocalDate taskDateTime = LocalDate
+                            .parse(splitInput[splitInput.length - 1]);
+                    return taskDateTime;
+                } catch (Exception e) {
+                    System.out.println("failed to read date");
+                }
+            }
+        }
+        return null;
+    }
```

Comment:
should try to avoid [deep nesting](https://nus-cs2103-ay2223s2.github.io/website/se-book-adapted/chapters/codeQuality.html#avoid-deep-nesting) or arrowhead-shaped code

would suggest you put the invalid cases at the top, then put the valid case at the bottom for better readability

```suggestion
        if (splitInput.length < 4) {
            // invalid case: <description>
            return null;
        } else if (!splitInput[splitInput.length - 2].equals("/by")) {
            // invalid case: <description>
            return null;
        } else {
            // do the thing
            ...
        }
```